Content.Server/Fluids/EntitySystems/AbsorbentSystem.Footprints.cs

```diff
@@ -0,0 +1,43 @@
+using System.Linq;
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Fluids;
+using Content.Shared.FootPrint;
+using Content.Shared.Timing;
+
+namespace Content.Server.Fluids.EntitySystems;
+
+public sealed partial class AbsorbentSystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+
+    /// <summary>
+    ///     Tries to clean a number of footprints in a range determined by the component. Returns the number of cleaned footprints.
+    /// </summary>
+    private int TryCleanNearbyFootprints(EntityUid user, EntityUid used, EntityUid target, AbsorbentComponent component,  Entity<SolutionComponent> absorbentSoln)
+    {
+        var footprintQuery = GetEntityQuery<FootPrintComponent>();
+        var targetCoords = Transform(target).Coordinates;
+        var entities = _lookup.GetEntitiesInRange(targetCoords, component.FootprintCleaningRange, LookupFlags.Uncontained);
```

Comment:
I'm pretty sure you can look up only for the entities with FootPrintComponent
`GetEntitiesInRange<FootPrintComponent>`